blockchain-albatross/src/blockchain/blockchain.rs

```diff
@@ -430,9 +309,33 @@ impl Blockchain {
             return Err(PushError::InvalidSuccessor);
         }
 
-        // Check if a view change occurred - if so, validate the proof
+        // Check that the current block timestamp is equal or greater than the timestamp of the
+        // previous block. This is necessary to guarantee that time is monotonically increasing.
+        // Having time go back could create issues with the blockchain state.
+        if prev_info.head.timestamp() >= header.timestamp() {
+            warn!("Rejecting block - block timestamp precedes parent timestamp");
+            return Err(PushError::InvalidSuccessor);
+        }
+
+        // Check that the current block timestamp less the node's system time is less than or equal
+        // to the allowed maximum drift. Basically, we check that the block isn't from the future.
+        // This is necessary so that validators cannot create blocks far in the future, which would
+        // give them bigger rewards.
+        // Both times are given in Unix time standard.
+        let now = SystemTime::now()
+            .duration_since(SystemTime::UNIX_EPOCH)
+            .expect("System time is before Unix epoch")
+            .as_millis() as u64;
```

Comment:
We usually want to rely on the NetworkTime instead of our local SystemTime.

Reply:
Agree with you there. But NetworkTime isn't implemented yet. We can merge this one now and then I'll do a PR for the NetworkTime later (which will be a lot bigger anyway).